app/assets/javascripts/initializers/initScrolling.js

```diff
@@ -1,26 +1,45 @@
 /* global insertAfter, insertArticles, buildArticleHTML, nextPage:writable, fetching:writable, done:writable, InstantClick */
 
-var client;
+// The purpose of initScrolling is to start paginating the page once you reach the bottom of the page.
+// It does this for majority of the views on the dashboard page as well as the home feed.
 
+var client;
+/**
+ *
+ * @param {String} el is an HTML element that contains the container that we want to append items to. It contains id: indexContainer.
+ * @param {String} endpoint is the relative URL of the endpoint to call to get the data
+ * @param {Function} insertCallback is a function from the return of the callback function (insertNext). The function is passed the data as an argument to build the HTML.
+ * @returns nil but calls the insertCallback (i.e. the insertEntries function) with the data as an argument to update the HTML. It also handles the loading spinner.
+ */
 function fetchNext(el, endpoint, insertCallback) {
   var indexParams = JSON.parse(el.dataset.params);
-  var urlParams = Object.keys(indexParams)
+  // I change the name of the param from "action" to "controller_action" prior to the fetch because Rails ignores the
+  // "action" param in the corresponding endpoint controller because it is a reserved word in Rails. Based on the above
+  // I thought it would be safe to do the param name update for now until we can refactor this file and change the coupled architecture.
+  // "action" renamed to "controller_action" is the name of the action from the originating server side request e.g. hidden_tags in the dashboard_controller.
+  const updatedIndexParams = {};
+  if (indexParams['action']) {
+    delete Object.assign(updatedIndexParams, indexParams, {
+      ['controller_action']: indexParams['action'],
+    })['action'];
+  }
+
+  var urlParams = Object.keys(updatedIndexParams)
     .map(function handleMap(k) {
-      return encodeURIComponent(k) + '=' + encodeURIComponent(indexParams[k]);
+      return (
+        encodeURIComponent(k) + '=' + encodeURIComponent(updatedIndexParams[k])
+      );
     })
     .join('&');
   if (urlParams.indexOf('q=') > -1) {
     return;
   }
```

Comment:
I've really tried not to change too much code that was in place already because there's just so much to refactor and without tests in place the potential to break things are higher. Also, I'd like to keep the refactors scoped in a separate issue. I'm keeping a list of notes that I'll add to an issue.

Reply:
I decided to test and address this!